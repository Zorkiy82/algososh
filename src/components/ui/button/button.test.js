import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./button";

describe("Тесты компонента Button", () => {
  function onClick() {
    alert("it work");
  }
  window.alert = jest.fn();

  it("Кейс #1 (без текста / без diasbled / без лоадера) корректно отрисовывается + вызывается колбек при клике на кнопку", () => {
    render(<Button onClick={onClick} />);
    const btn = screen.getByTestId("button-id");
    expect(btn).toHaveTextContent("");
    expect(btn).toMatchSnapshot();

    fireEvent.click(btn);
    expect(window.alert).toHaveBeenCalledWith("it work");
  });

  it("Кейс #2 (c текстом) корректно отрисовывается + вызывается колбек при клике на кнопку", () => {
    render(<Button text="Test" onClick={onClick} />);
    const btn = screen.getByTestId("button-id");
    expect(btn).toHaveTextContent("Test");
    expect(btn).toMatchSnapshot();

    fireEvent.click(btn);
    expect(window.alert).toHaveBeenCalledWith("it work");    
  });

  it("Кейс #3 (c diasbled) корректно отрисовывается + НЕ вызывается колбек при клике на кнопку", () => {
    const { rerender } = render(<Button onClick={onClick} />);
    const btn = screen.getByTestId("button-id");
    expect(btn).not.toHaveAttribute("disabled");
    expect(btn).toMatchSnapshot();

    fireEvent.click(btn);
    expect(window.alert).toHaveBeenCalledWith("it work");

    rerender(<Button onClick={onClick} disabled />);
    expect(btn).toHaveAttribute("disabled");
    expect(btn).toMatchSnapshot();

    fireEvent.click(btn);
    expect(window.alert).toBeCalledTimes(1);
  });

  it("Кейс #4 (кнопка с лоадером) корректно отрисовывается + НЕ вызывается колбек при клике на кнопку", () => {
    const { rerender } = render(<Button onClick={onClick} />);
    const btn = screen.getByTestId("button-id");    
    expect(screen.queryByTestId("loader-id")).toBeNull();
    expect(btn).toMatchSnapshot();

    fireEvent.click(btn);
    expect(window.alert).toHaveBeenCalledWith("it work");

    rerender(<Button  isLoader={true} onClick={onClick} />);
    expect(screen.queryByTestId("loader-id")).toBeInTheDocument();
    expect(btn).toMatchSnapshot();

    fireEvent.click(btn);
    expect(window.alert).toBeCalledTimes(1);
  });
});
