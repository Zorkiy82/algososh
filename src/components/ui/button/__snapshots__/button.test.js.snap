// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тесты компонента Button Кейс #1 (без текста / без diasbled / без лоадера) корректно отрисовывается + вызывается колбек при клике на кнопку 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="button-id"
  type="button"
>
  <p
    class="text undefined"
    data-testid="text-id"
  />
</button>
`;

exports[`Тесты компонента Button Кейс #2 (c текстом) корректно отрисовывается + вызывается колбек при клике на кнопку 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="button-id"
  type="button"
>
  <p
    class="text undefined"
    data-testid="text-id"
  >
    Test
  </p>
</button>
`;

exports[`Тесты компонента Button Кейс #3 (c diasbled) корректно отрисовывается + НЕ вызывается колбек при клике на кнопку 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="button-id"
  type="button"
>
  <p
    class="text undefined"
    data-testid="text-id"
  />
</button>
`;

exports[`Тесты компонента Button Кейс #3 (c diasbled) корректно отрисовывается + НЕ вызывается колбек при клике на кнопку 2`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="button-id"
  disabled=""
  type="button"
>
  <p
    class="text undefined"
    data-testid="text-id"
  />
</button>
`;

exports[`Тесты компонента Button Кейс #4 (кнопка с лоадером) корректно отрисовывается + НЕ вызывается колбек при клике на кнопку 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="button-id"
  type="button"
>
  <p
    class="text undefined"
    data-testid="text-id"
  />
</button>
`;

exports[`Тесты компонента Button Кейс #4 (кнопка с лоадером) корректно отрисовывается + НЕ вызывается колбек при клике на кнопку 2`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  data-testid="button-id"
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    data-testid="loader-id"
    src="loader.svg"
  />
</button>
`;
